lab-7-evan/lib/body-parser.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+module.exports = function(req, callback) {
+  req.body = '';
+
+  req.on('data', function(data) {
+    req.body += data.toString();
+  });
+
+  req.on('end', function() {
+    try {
+      req.body = JSON.parse(req.body);
```

Comment:
You need to call your callback when the data is done being parsed to JSON. Remember how we call a callback with no error and some data.